### ISSUE TITLE
Influx 1.0.2 memory consumption spikes 

### ISSUE THREAD

**Issue by rtv2222** [influxdata/influxdb#7508]
### Bug report

Influx DB Version : 1.0.2
OS : Ubuntu 14.04 on VMware
cpus : 8
Memory : 16 GB
Disk : SSD

**Steps to reproduce:**

I have 4 measurements on influx with the largest of them having around 3086448 points. Data is being written to this measurement at the rate of 4000 points every minute.

After about 2 hours of uptime, InfluxDB memory usages spikes from 1.2 GB to 16 GB. The spikes are about 30 mins apart and while the consumption comes down eventually, influxdb freezes and stops responding to any requests (via InfluxDB java client or over http). Influx would have to be killed (kill -9) and restart for it to start responding to queries again.

Here is the HeapAloc output from the _internal database of Influx as seen on Chronograph.

![image](https://cloud.githubusercontent.com/assets/13368028/19655455/96caf132-9a39-11e6-8367-91e9401ed194.png)

**Expected behavior:** Continue to use around 2 GB of memory instead of the spike. Continue responding to queries.

**Actual behavior:**  Please see the description above

**Additional info:** 

I am already seeing spikes 10 mins into using 1.0.2 version. Please let me know if this helps. I can continue to update this case as I see more spikes. 

[shards.txt](https://github.com/influxdata/influxdb/files/548490/shards.txt)
[block.txt](https://github.com/influxdata/influxdb/files/548489/block.txt)
[diagnostics.txt](https://github.com/influxdata/influxdb/files/548492/diagnostics.txt)
[goroutine.txt](https://github.com/influxdata/influxdb/files/548491/goroutine.txt)
[heap.txt](https://github.com/influxdata/influxdb/files/548494/heap.txt)
[iostat.txt](https://github.com/influxdata/influxdb/files/548493/iostat.txt)
[stats.txt](https://github.com/influxdata/influxdb/files/548496/stats.txt)
[vars.txt](https://github.com/influxdata/influxdb/files/548495/vars.txt)

Here is the heap dump when htop reported 8 Gig consumption by influx

[heap.txt](https://github.com/influxdata/influxdb/files/548525/heap.txt)

**Comment by jwilder** [comment]
@rtv2222 Memory usage could spike due to queries and other background task running on the system.  The the system will not stay at 2GB if there are writes and queries running because memory usage is dependent on your workload.   What queries are you running and what CQs do you have configured?  Also, you desc mentions that the system is frozen, is this still occurring?  

You'll need to grab a heap profile during one of those larger spikes for us to see what is using memory.

**Comment by rtv2222** [comment]
@jwilder  - I fully understand that memory usage could spike during queries. Its the amount of spike that worries me. I see it spiking to 9 Gig on running a moderately complex query. CQs run once every hour on this system. I have also updated a heap dump when the memory usage went up to 7 Gig while running the following queries. CQs were not running when this spike occurred.

[heap.txt](https://github.com/influxdata/influxdb/files/548547/heap.txt)

Here is one of the queries that causes a spike
select mean(cpu) as "mean_cpu", mean(disk) as "mean_disk", mean(memory) as "mean_memory", mean(network) as "mean_network" from caf_metrics where time > now() - 1h group by serial_id;select mean(mean_cpu) as "mean_cpu", mean(mean_disk) as "mean_disk", mean(mean_memory) as "mean_memory", mean(mean_network) as "mean_network" from appmgr_metrics.am_1d_cq_rp.aggregate_metrics_1d where time > now() - 1d group by serial_id

and here is another
select max(memory) from caf_metrics where time > now() - 1h  and app_id = 'dna_1477113196266' group by serial_id;select max(max_memory) from appmgr_metrics.am_1d_cq_rp.aggregate_metrics_1d where time > now() - 1d and app_id = 'dna_1477113196266' group by serial_id
--
select max(cpu) from caf_metrics where time > now() - 1h  and app_id = 'dna_1477113196266' group by serial_id;select max(max_cpu) from appmgr_metrics.am_1d_cq_rp.aggregate_metrics_1d where time > now() - 1d and app_id = 'dna_1477113196266' group by serial_id
--
select max(cpu),min(cpu),mean(cpu) from caf_metrics where time > now() - 1d and app_id = 'dna_1477113196266' and cpu >= 0 group by time(1h)  fill(0)
--
select max(disk),min(disk),mean(disk) from caf_metrics where time > now() - 1d and app_id = 'dna_1477113196266' and disk >= 0 group by time(1h)  fill(0)
--
select max(memory),min(memory),mean(memory) from caf_metrics where time > now() - 1d and app_id = 'dna_1477113196266' and memory >= 0 group by time(1h)  fill(0)
--
select max(network),min(network),mean(network) from caf_metrics where time > now() - 1d and app_id = 'dna_1477113196266' and network >= 0 group by time(1h)  fill(0)

The freeze has not occurred yet. I have 16 Gig of RAM on my system. I saw the freeze with 0.12.2 when almost all of the available memory was used up by influx. I will let you know how this version behaves. The curl requests for heap will not work during that time. Do let me know how I must obtain the heap when influx does not respond to curl requests. 

Thanks,
Rajesh
